collector/src/main/java/org/apache/hertzbeat/collector/collect/common/cache/ConnectionCommonCache.java

```diff
@@ -197,15 +197,18 @@ public Optional<C> getCache(T key, boolean refreshCache) {
             cacheMap.remove(key);
             return Optional.empty();
         }
-        C value = cacheMap.get(key);
-        if (value == null) {
-            log.error("[connection common cache] value is null, remove it, key {}.", key);
-            cacheMap.remove(key);
```

Comment:
```java
cacheMap.remove(key);
```

Why was this line of code deleted?

Reply:
Because `cacheMap` uses `ConcurrentLinkedHashMap<K, V>`. The implementation of `put()` method in `ConcurrentLinkedHashMap<K, V>` checks for null value and throws `NullPointerException` if value is null. So `cache.get(key) == null` is true only when the key does not exist in the map.

Reply:
Please consider upgrading from CLHM to Caffeine at some opportune time. Thanks!

Reply:
> Please consider upgrading from CLHM to Caffeine at some opportune time. Thanks!

Thanks, we will upgrade to Caffeine in the near future！